### ISSUE TITLE
Downloaded file loses original dates

### ISSUE THREAD

**Issue by ansiblejunky** [marlowintexas/ApplePhotoProgram#1]
Thanks for this great script, but I just couldn't use it because when you download the files locally it loses the capture/create date. All of the files downloaded now have the same file create/modified date of the time when it was downloaded. Another tool does it correctly: 
https://github.com/icloud-photos-downloader/icloud_photos_downloader

But that tool doesn't do the date range like you do, which I love that part. So was hoping you can fix this? 

Thanks,
John

**Comment by marlowintexas** [comment]
Let me check that in the coming days.  I have updates that script awhile
back and need to take another look.

David

**Comment by ansiblejunky** [comment]
That's awesome! I need to use a script like this each year to download last year's photos so my wife can create a "year in review" video she puts together for our family. haha Also, to simply store my photos on a drive without needing Apple's iCloud.

**Comment by marlowintexas** [comment]
I looked at my photos and the digital Exif date is correct to the original
shoot date of the photo, but the "create" file system date defaults to when
it was added to the file system.  I usually like to keep my files set to
the create/modify date on the file system and not re-touch the file to the
original shoot date of the photo.  Let me think on that a little bit more
on if I want that to change or not.  I should be able to execute a touch to
change the date of the file though.  Keep in mind, the reason behind my
original program was to just have a backup of what was on iCloud in case
something went wrong or goofed up.  It was not to be used to manage back
and forth per se.  I wanted to make the icloud the main place for my
family's photos.

I will say a bit has changed on the program since I put that up on Git so I
need to do some work to get the new program up there.

David

**Comment by ansiblejunky** [comment]
Yeah I understand that reasoning. For me, even though the files are organized in sub folders, the files listed in Finder show them all created and modified today. One could argue that the file has not been changed, and is just a backup, and therefore should hold original creation/mod dates as well. 

As any good software, you can elect to offer a parameter to enable this or not. That's the ideal way to implement it.

**Comment by ansiblejunky** [comment]
In addition, was just thinking about this more... When I use my backup software (Carbon Copy Cloner) to backup folders etc, it retains the create/mod dates on files. However, in contrast, when I use something rudimentary like `rsync` to backup files, it results in all the backed up files having today's date for create/mod timestamps. That's definitely not good for me, as I use that timestamp for a lot of things with documents etc.

**Comment by jaimetur** [comment]
Nice and very simple script. 
Did you finally solved the filesystem creation/modification date to be aligned with exif info?

As per your comments you have an updated version of this script. It would be nice if you could upload it or share it with me. 

Many many thanks in advance.

**Comment by marlowintexas** [comment]
I have not.  I need to get back to it some day.  My apologies…
